### PR TITLE
Consul nameresolution component - runtime changes

### PR DESCRIPTION
**Pull request by a-elsheikh**
# Description

This PR contains the supporting changes for the new consul nameresolution component. This also includes the ability to configure the target nameresolution component and parse environment variables from config in the runtime.

## Issue reference

dapr/components-contrib#749

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/runtime/runtime.go [hunk 1]
```diff
@@ -1064,34 +1064,51 @@ func (a *DaprRuntime) initNameResolution() error {
...
+			resolverName = "mdns"
+		default:
+			return errors.Errorf("unable to determine name resolver for %s mode", string(a.runtimeConfig.Mode))
 		}
-	default:
-		return errors.Errorf("remote calls not supported for %s mode", string(a.runtimeConfig.Mode))
+	}
+
+	if resolverVersion == "" {
+		resolverVersion = "v1"
```
Comment (yaron2) [inline pkg/runtime/runtime.go:1082]: Can `v1` be a constant somewhere?

**Review by yaron2 — CHANGES_REQUESTED** [review]

File: pkg/runtime/runtime.go (hunk 1 above)
Comment (a-elsheikh) [reply pkg/runtime/runtime.go]: done, have a look

**Comment by pkedy** [comment]
@a-elsheikh I got this working in Kubernetes but I had to make some changes. Since I don't have access to push to your branch, I'm attaching this diff/patch. [consul_patch.diff.txt](https://github.com/dapr/dapr/files/6297509/consul_patch.diff.txt). Please take a look. Thanks!

**Comment by a-elsheikh** [comment]
> @a-elsheikh I got this working in Kubernetes but I had to make some changes. Since I don't have access to push to your branch, I'm attaching this diff/patch. [consul_patch.diff.txt](https://github.com/dapr/dapr/files/6297509/consul_patch.diff.txt). Please take a look. Thanks!

Nice! I've applied it and pushed

**Comment by yaron2** [comment]
Lets merge the contrib PR and update the branch here so we can run e2e.

/cc @pkedy

**Comment by yaron2** [comment]
@a-elsheikh we merged the components-contrib PR.

Can you please make the following changes?

1. Run `go get -u github.com/dapr/components-contrib@master`
2. Run `go mod tidy`
3. Push your changes

**Comment by a-elsheikh** [comment]
> @a-elsheikh we merged the components-contrib PR.
> 
> Can you please make the following changes?
> 
> 1. Run `go get -u github.com/dapr/components-contrib@master`
> 2. Run `go mod tidy`
> 3. Push your changes

done

**Review by yaron2 — APPROVED** [review]

**Comment by yaron2** [comment]
@a-elsheikh the only thing left to do now is open a docs issue and PR to explain how to configure and use the Consul component.

**Comment by a-elsheikh** [comment]
> @a-elsheikh the only thing left to do now is open a docs issue and PR to explain how to configure and use the Consul component.

yep, that already exists - dapr/docs#1408

**Comment by pkedy** [comment]
Congratulations @a-elsheikh! I know this was a long time coming. Apologies for the delay.

**Comment by a-elsheikh** [comment]
> Congratulations @a-elsheikh! I know this was a long time coming. Apologies for the delay.

@pkedy @yaron2 thank you both for your guidance, I'll start having a look at the docs now 😄

**Pull request merged**